aggregator/src/aggregator.rs

```diff
@@ -1430,6 +1502,35 @@ impl VdafOps {
                     )
                     .get_encoded(),
                 )
+            };
+
+            let check_keypairs = if task_hpke_keypair.is_none() && global_hpke_keypair.is_none() {
+                info!(
+                    config_id = %report_share.encrypted_input_share().config_id(),
+                    "Helper encrypted input share references unknown HPKE config ID"
+                );
+                aggregate_step_failure_counter.add(
+                    &Context::current(),
+                    1,
+                    &[KeyValue::new("type", "unknown_hpke_config_id")],
+                );
+                Err(ReportShareError::HpkeUnknownConfigId)
+            } else {
+                Ok(())
+            };
+
+            let plaintext = check_keypairs.and_then(|_| {
+                match (task_hpke_keypair, global_hpke_keypair) {
+                    (None, None) => unreachable!("already checked this condition"),
+                    (None, Some(global_hpke_keypair)) => try_hpke_open(&global_hpke_keypair),
+                    (Some(task_hpke_keypair), None) => try_hpke_open(task_hpke_keypair),
+                    (Some(task_hpke_keypair), Some(global_hpke_keypair)) => {
+                        match try_hpke_open(task_hpke_keypair) {
+                            Ok(result) => Ok(result),
+                            Err(_) => try_hpke_open(&global_hpke_keypair),
```

Comment:
Similar to above, let's check what the error is before we fall back to the global key.